src/teleop_twist_joy.cpp

```diff
@@ -68,63 +68,73 @@ struct TeleopTwistJoy::Impl
 
 /**
  * Constructs TeleopTwistJoy.
- * \param nh NodeHandle to use for setting up the publisher and subscriber.
- * \param nh_param NodeHandle to use for searching for configuration parameters.
  */
-TeleopTwistJoy::TeleopTwistJoy(rclcpp::Node::SharedPtr & node)
+TeleopTwistJoy::TeleopTwistJoy() : Node("teleop_twist_joy_node")
 {
-  pimpl_ = new Impl();
+  pimpl_ = new Impl;
 
-  pimpl_->node = node;
-
-  pimpl_->cmd_vel_pub = node->create_publisher<geometry_msgs::msg::Twist>("cmd_vel",
+  pimpl_->cmd_vel_pub = this->create_publisher<geometry_msgs::msg::Twist>("cmd_vel",
     rmw_qos_profile_sensor_data);
-  pimpl_->joy_sub = node->create_subscription<sensor_msgs::msg::Joy>("joy",
+  pimpl_->joy_sub = this->create_subscription<sensor_msgs::msg::Joy>("joy",
     std::bind(&TeleopTwistJoy::Impl::joyCallback, this->pimpl_, std::placeholders::_1),
     rmw_qos_profile_sensor_data);
 
-  pimpl_->enable_button = 5;
-  node->get_parameter("enable_button", pimpl_->enable_button);
-
-  pimpl_->enable_turbo_button = -1;
-  node->get_parameter("enable_turbo_button", pimpl_->enable_turbo_button);
-
-  // TODO(clalancette): node->get_parameter(s) doesn't seem to
-  // support getting a map of values yet.  Revisit this once it does.
-  // if (nh_param->getParam("axis_linear", pimpl_->axis_linear_map))
-  // {
-  //   nh_param->getParam("axis_linear", pimpl_->axis_linear_map);
-  //   nh_param->getParam("scale_linear", pimpl_->scale_linear_map);
-  //   nh_param->getParam("scale_linear_turbo", pimpl_->scale_linear_turbo_map);
-  // }
-  // else
-  {
-    pimpl_->axis_linear_map["x"] = 5;
-    node->get_parameter("axis_linear", pimpl_->axis_linear_map["x"]);
-    pimpl_->scale_linear_map["x"] = 0.5;
-    node->get_parameter("scale_linear", pimpl_->scale_linear_map["x"]);
-    pimpl_->scale_linear_turbo_map["x"] = 1.0;
-    node->get_parameter("scale_linear_turbo", pimpl_->scale_linear_turbo_map["x"]);
+  // TODO(clalancette): node->get_parameter(s) doesn't support getting a map of
+  // values yet.  Revisit this once it does.
+  int_parameters.push_back({"enable_button", 5, pimpl_->enable_button});
+  int_parameters.push_back({"enable_turbo_button", -1, pimpl_->enable_turbo_button});
+  int_parameters.push_back({"axis_linear_x", 5, pimpl_->axis_linear_map["x"]});
+  int_parameters.push_back({"axis_angular_yaw", 2, pimpl_->axis_angular_map["yaw"]});
+
+  double_parameters.push_back({"scale_linear_x", 0.5, pimpl_->scale_linear_map["x"]});
+  double_parameters.push_back({"scale_linear_turbo_x", 1.0, pimpl_->scale_linear_turbo_map["x"]});
+  double_parameters.push_back({"scale_angular_yaw", 0.5, pimpl_->scale_angular_map["yaw"]});
+  double_parameters.push_back({"scale_angular_turbo_yaw", 0.5, pimpl_->scale_angular_turbo_map["yaw"]});
+
+  for(const auto& p : int_parameters) {
+    this->set_parameter_if_not_set(p.name, p.default_value);
+    this->get_parameter(p.name, p.variable);
```

Comment:
Could this use `get_parameter_or` ?

Reply:
See https://github.com/ros2/teleop_twist_joy/pull/8#discussion_r207889563; unless there is something I'm missing, I don't think this will accomplish the same thing.